### ISSUE TITLE
Publish to maven central

### ISSUE THREAD

**Issue by krzyk** [couchbase/CouchbaseMock#16]
I can't find this project in maven central, please point me to the correct place or publish it to maven central.
Without this it won't take much traction as most of companies don't allow downloading artifacts from other repositories.

**Comment by mmoayyed** [comment]
:+1:

**Comment by mmoayyed** [comment]
I would want to write integration tests against this mock, and I would need to find a way to start the mock server programmatically, which would mean that I would need access to binary artifacts and APIs presented in this mock ,and I would rather not compile and built this locally.

**Comment by oliverlietz** [comment]
Same here for [Apache Sling](https://sling.apache.org): [SlingLaunchpadNosqlCouchbaseIT](https://github.com/apache/sling/blob/trunk/contrib/launchpad/karaf/org.apache.sling.launchpad.karaf-integration-tests/src/test/java/org/apache/sling/launchpad/karaf/tests/bootstrap/SlingLaunchpadNosqlCouchbaseIT.java)

**Comment by mnunberg** [comment]
@ingenthr - what are your thoughts?

**Comment by Thib17** [comment]
I also need this project to be available on maven central. Is it possible to publish it on maven central ? I can help, if you need.

**Comment by ingenthr** [comment]
I have no problems with publishing it there, just recognizing that it's a project, not a product and so we're a little looser about interfaces on versions.  @mnunberg - you can pull @simonbasle in to get it under the right account and such if it helps.

**Comment by ingenthr** [comment]
I just checked, and the only thing that worries me is we're calling it 1.4.2.  Since semver is our standard for product in libraries from Couchbase, I'd feel a bit not good if we break something.  

For a maven central publishing it seems like we could do one of two things:
1) Update the docs/README to point out the project/interface stability expected here, or 
2) Change the version back to an 0.<something> which signals interface stability expectations

I'd rather do the second one.  That breaks a few people-- mainly us.  What do you think @mnunberg?

**Comment by mnunberg** [comment]
We do use semver, however semver here represents the public interface as far as the mock is used as a standalone server -- no guarantee at all is made about the Java APIs; so 1.x.x is "protocol compatible" with any other 1.x version; 1.1, 1.2, 1.3, etc have all had features added to them.

Adding a '0' might be confusing considering that there's an expectation that a 1.x is newer than a 0.x; and it would be rather odd to "upgrade" from 1.4.2 to 0.4.3. Additionally, the prior version was 0.8, which was essentially a non-sensical version.

I'm partial to maintaining our versioning as-is, and assuming that people will test their code before upgrading -- after all, we're simply publishing on maven central as a matter of convenience so users won't need to tote around a bunch of jars.

**Comment by nikolayspb** [comment]
So is it still not published?

**Comment by ingenthr** [comment]
It is not at the moment @nikolayspb.  For the most part, this is a testing tool for our SDKs, so it's not something that has been a high priority, as you can tell.

That said, @mnunberg, do you think we can do this?  Let's just document our current interface stability standard via javadoc and README.

**Comment by jimhooker2002** [comment]
Hi @ingenthr - just wanted to provide an "outsiders" perspective here in order to make sure it gets heard (I appreciate there's no pushback on getting this project into Maven, although it is taking a surprising amount of time to make that happen).  

Specifically, wanted to add some commentary to this statement: "For the most part, this is a testing tool for our SDKs".

Even though it started as a testing tool for your SDKs, specifically as an app you start-up separate from the code under test, it can also be run in "embedded mode" as is obvious from the unit tests.

Here's the "outsiders" perspective:  that embedded mode is a huge deal and a tool that many of us are clamouring for.  If you guys aren't already taking it seriously, and actively trying to support that use-case, then hopefully I can convince you that you should.

Today, folks are used to using tools such as HSQL or [DynamoDBLocal](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html), both of which can run in "embedded mode", in order to test their software.

That style of testing allows us to test more of the stack, earlier on in our development cycle.  It's a real boon as far as quality and productivity is concerned.

By providing such a tool, you add one more reason to use Couchbase.  You make the whole Couchbase eco-system that little bit better.  Of course, the better that eco-system, the greater adoption will be (and we all want that).

So, in summary - embedded mode is a huge deal.  I'm hoping that minds will be shifted in that direction and that the product and the documentation will reflect accordingly.

I'm really happy to help contribute here however I can.  CouchbaseMock is an incredibly useful tool.  If I can do anything to move us forward in terms of the "embedded" vision, then please let me know.  Perhaps I can help with this ticket specifically?  How can I contribute to getting this into Maven?

**Comment by qwwdfsad** [comment]
Hi, I understand it's not in your priority list, but nevertheless it's a very useful feature, e.g. for testing your own wrappers around reactive API.
@ingenthr Is there anything external contributors can do to make this release happen?

**Comment by mnunberg** [comment]
We're one step closer to pushing on Maven Central. We just changed the internal package names from `org.couchbase` to `com.couchbase` since we own the latter in Maven.

**Comment by mnunberg** [comment]
https://search.maven.org/#artifactdetails%7Ccom.couchbase.mock%7CCouchbaseMock%7C1.5.1%7Cjar
